addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -158,12 +158,10 @@ def _generate_efaktur_invoice(self, delimiter):
 
             street = ', '.join([x for x in (move.partner_id.street, move.partner_id.street2) if x])
 
-            invoice_npwp = '000000000000000'
-            if commercial_partner.vat and len(commercial_partner.vat) >= 12:
-                invoice_npwp = commercial_partner.vat
-            elif (not commercial_partner.vat or len(commercial_partner.vat) < 12) and commercial_partner.l10n_id_nik:
-                invoice_npwp = commercial_partner.l10n_id_nik
-            invoice_npwp = invoice_npwp.replace('.', '').replace('-', '')
+            invoice_npwp = commercial_partner.vat.replace('.', '').replace('-', '')
```

Comment:
what happens if the partner does not have a vat set? Before we would still support it, but now we wouldn't?

Reply:
Good point you've raised, I've talked with JGR about this
previously in Odoo: if user fill in VAT and it's legit use it. If no VAT, use NIK if possible. If both none, then use '000000000000000'
current: we dont' want to auto-assign this '000000000000000'. And now since NPWP (VAT) and NIK is the same thing due to regulation change, they would have to fill in VAT. So I was thinking, what if we raise error when VAT is not filled in?

Reply:
But what about users currently in 15.0 who didn't fill the vat because it works anyway? Suddenly, their customer configuration will be wrong even though it used to work.

Just asking, we need to make sure to impact the end user the least we can if it is possible and makes sense.

Reply:
discussed with JGR again, she said that people are expected to fill in the VAT even with '00000000000' before generating efaktur. So I think we can proceed with raising error, and in the error probably mention something like "Please fill in the customer's NPWP before generating e-Faktur"